tests/error/misc_errors/conditional_global_shadow.err

```diff
@@ -0,0 +1,14 @@
+Error: Different types (at $FILE:14:11)
+   | 
+12 |     if b:
+13 |         x = 4
+14 |     return x
+   |            ^ Variable `x` may refer to different types
+   | 
+13 |         x = 4
+   |         - This is of type `int`
+
+Note: Variable `x` may be shadowing a global name with type `() -> None` on some
+branches
```

Comment:
This message reads a bit akward, but hard to improve without cross-file diagnostics... :/